Tests/BSONTests/ExtendedJSONConversionTests.swift

```diff
@@ -4,7 +4,75 @@ import Nimble
 import NIO
 import XCTest
 
+extension CodingUserInfoKey {
+    static let barInfo = CodingUserInfoKey(rawValue: "bar")!
+}
+
 open class ExtendedJSONConversionTestCase: BSONTestCase {
+    struct Test: Codable, Equatable {
+        let x: Bool
+        let y: Int32
+        let z: BSONRegularExpression
+    }
+
+    struct Foo: Codable, Equatable {
+        let foo: Bool
+
+        enum CodingKeys: String, CodingKey {
+            case foo
+        }
+
+        func encode(to encoder: Encoder) throws {
+            let barInfo = encoder.userInfo[.barInfo] as! Bool
```

Comment:
since we have the explicit `== true` below, I think we can drop this force cast to prevent the test runner from crashing when this fails.

Reply:
done